doc/man3/SRP_VBASE_new.pod

```diff
@@ -0,0 +1,73 @@
+=pod
+
+=head1 NAME
+
+SRP_VBASE_new,
+SRP_VBASE_free,
+SRP_user_pwd_free,
+SRP_VBASE_init,
+SRP_VBASE_get1_by_user,
+SRP_VBASE_get_by_user
+- Functions to create and manage a stack of SRP user authentication data
+
+=head1 SYNOPSIS
+
+ #include <openssl/srp.h>
+
+ SRP_VBASE *SRP_VBASE_new(char *seed_key);
+ void SRP_VBASE_free(SRP_VBASE *vb);
+ void SRP_user_pwd_free(SRP_user_pwd *user_pwd);
+
+ int SRP_VBASE_init(SRP_VBASE *vb, char *verifier_file);
+
+ SRP_user_pwd *SRP_VBASE_get1_by_user(SRP_VBASE *vb, char *username);
+ SRP_user_pwd *SRP_VBASE_get_by_user(SRP_VBASE *vb, char *username);
+
+=head1 DESCRIPTION
+
+The SRP_VBASE_new() function allocates a structure to store SRP authentication
+information. If B<seed_key> is not NULL a copy is stored and used to generate
+authentication parameters for unknown users.
+
+The SRP_VBASE_free() function frees up the B<vb> structure.
+If B<vb> is NULL, nothing is done.
+
+The SRP_user_pwd_free() function frees up the B<user_pwd> structure.
+If B<user_pwd> is NULL, nothing is done.
+
+The SRP_VBASE_init() function parses the information in a verifier file and
+populates the B<vb> structure.
```

Comment:
It's not very clear, looking quickly over the libsrp parsing code from [t_pw.c](https://github.com/secure-remote-password/stanford-srp/blob/587900d32777348f98477cb25123d5761fbe3725/libsrp/t_pw.c#L337) it seems to match the second format written in the comment at the top of [SRP_VBASE_init()](https://github.com/openssl/openssl/blob/91d2c83fe03d238270b9d07d8f336fae270afe5d/crypto/srp/srp_vfy.c#L380).

But the actual implementation of the function does none of that (unless I had a hard missread on it), see my previous comment.

Reply:
The first format seems to almost match this code from libsrp:

https://github.com/secure-remote-password/stanford-srp/blob/587900d32777348f98477cb25123d5761fbe3725/libsrp/t_conf.c#L550-L562

The only difference seems to be the trailing ":0" in the comment.

The stanford SRP site seems to have some old patches for OpenSSL from before they were integrated into the mainline:
http://srp.stanford.edu/download.html

These patches have exactly the same comment, and the implementation also does not seem to match the comment. So I conclude that this comment has been out of date since before this code even made it into OpenSSL master.

The format actually seems to be:

V base64(verifier) base64(salt) username gNid userinfo(optional)

It seems from code inspection that you can also have lines like this:

I base64(N) base64(g)

However there doesn't seem to be any support that I could find for actually *creating* such entries, so it is doubtful whether anyone is actually using this capability.

Finally it also seems possible to have an entry in the same format as for a user but using an "R" instead of a "V", to indicate that the user is revoked.

Note of course that "base64" means the special libsrp version of base64. Since we're not even using the same file format, it seems fairly pointless using this "special" base64. But we're probably stuck with it for backwards compat reasons.

We should update the comment I think!

Reply:
I added some details in the man page and the implementation comment but didn't write down the specifics of the base64 format since it doesn't have any direct use.